File: ovvium-backend/src/main/java/com/ovvium/services/web/controller/bff/v1/transfer/response/account/LoginResponse.java
package com.ovvium.services.web.controller.bff.v1.transfer.response.account;

import com.ovvium.services.web.controller.bff.v1.transfer.response.user.UserCustomerResponse;
import com.ovvium.services.web.controller.bff.v1.transfer.response.user.UserResponse;
import lombok.Data;

@Data
public final class LoginResponse {

	private final UserResponse user;
	private final SessionResponse session;

	public LoginResponse(UserResponse user, SessionResponse session) {
		this.user = user;
		this.session = session;
	}

	public LoginResponse(UserCustomerResponse userCustomerResponse, SessionResponse session) {
		this.user = userCustomerResponse;
		this.session = session;
	}

	public LoginResponse(UserResponse user) {
		this(user, null);
	}
}

File: ovvium-app/src/model/response/LoginResponse.ts
import { User } from './../User';
import { SessionResponse } from './SessionResponse';

export class LoginResponse {
    
    user: User;
    session: SessionResponse;

    constructor(loginResponse: LoginResponse = {} as LoginResponse){
        this.user = loginResponse.user;
        this.session = loginResponse.session;
    }

}
